### ISSUE TITLE
Plot and summarise sites and mutations per tree

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/tsinfer-qc#8]
```python
import numpy as np

tree_sites = np.zeros(tinfo.ts.num_trees)
tree_mutations = np.zeros(tinfo.ts.num_trees)

for tree in tinfo.ts.trees():
    tree_sites[tree.index] = tree.num_sites
    tree_mutations[tree.index] = tree.num_mutations
```

(Note tree.num_mutatinos seems slow, we may need to implement that differently here)

**Comment by jeromekelleher** [comment]
- plot_sites_per_tree_distribution
- plot_mutations_per_tree_distribution
- plot_sites_per_tree
- plot_mutations_per_tree

Last two are along the genome.

**Comment by benjeffery** [comment]
> (Note tree.num_mutatinos seems slow, we may need to implement that differently here)

Using the `breakpoints` to make a per-site array of tree indexes, then mapping this over to mutations would seem to be the quick way to do this (`breakpoints` is cached when `TreeSequence` is loaded):

```python
site_tree_index = np.searchsorted(ts.breakpoints(as_array=True), ts.sites_position, side='right') - 1
unique_values, counts = np.unique(site_tree_index, return_counts=True)
sites_per_tree = np.zeros(ts.num_trees, dtype=np.int64)
sites_per_tree[unique_values] = counts

mutation_tree_index = site_tree_index[ts.mutations_site]
unique_values, counts = np.unique(mutation_tree_index, return_counts=True)
mutations_per_tree = np.zeros(ts.num_trees, dtype=np.int64)
mutations_per_tree[unique_values] = counts
```

**Comment by jeromekelleher** [comment]
Nice - I didn't think of this!

**Comment by savitakartik** [comment]
Closed in pr #24
